src/Tickets/Emails/Email_Abstract.php

```diff
@@ -122,6 +131,34 @@ public function hook() {
 		$this->set_placeholders( $default_placeholders );
 	}
 
+	/**
+	 * Sets the Dispatcher instance used by this email.
+	 * If no dispatcher instance was passed it will the default one for Tickets Emails.
+	 *
+	 * @since TBD
+	 *
+	 * @param Dispatcher|null $dispatcher
+	 */
+	public function set_dispatcher( ?Dispatcher $dispatcher = null ): void {
+		if ( null === $dispatcher ) {
+			$dispatcher = new Dispatcher;
+			$dispatcher->set_email( $this );
+		}
+
+		$this->dispatcher = $dispatcher;
+	}
+
+	/**
+	 * Gets the defined instance of the dispatcher for this email.
+	 *
+	 * @since TBD
+	 *
+	 * @return Dispatcher
+	 */
+	public function get_dispatcher(): Dispatcher {
+		return $this->dispatcher;
```

Comment:
Do we want a check here to be sure we're returning something useful? Call `set_dispatcher()` if not?

Reply:
If we do that, the emails don't have to `set` the Dispatcher just `get` it, unless they are using a specific one.

Reply:
Not in this case, it's part of the building of the method, I would rather always  check for if dispatcher is not null. Need to change the signature of the method to be `?Dispatcher`